web/src/components/UploadLicenseFile.tsx

```diff
@@ -0,0 +1,683 @@
+        <div className="flex u-marginTop--15 alignItems--center">
+          <span className="icon restore-icon" />
+          <Link
+            className="u-fontSize--normal u-linkColor u-fontWeight--medium u-textDecoration--underlineOnHover u-marginRight--5"
+            to="/restore"
+          >
+            {`Restore ${
+              applicationName ? `${applicationName}` : "app"
+            } from a snapshot`}{" "}
+          </Link>
+          <Icon icon="next-arrow" style={{ marginTop: "2px" }} size={9} />
+        </div>
+      )}
+      {isBackupRestore && snapshot ? (
+        <button
+          className="btn primary u-marginTop--20"
+          onClick={() => startRestore(snapshot)}
+          disabled={!hasFile}
+        >
+          {" "}
+          Start restore{" "}
+        </button>
+      ) : null}
+
+      <Modal
+        isOpen={viewErrorMessage}
+        onRequestClose={toggleViewErrorMessage}
+        contentLabel="Online install error message"
+        ariaHideApp={false}
+        className="Modal"
+      >
+        <div className="Modal-body">
+          <div className="ExpandedError--wrapper u-marginTop--10 u-marginBottom--10">
+            <p className="u-fontSize--small u-fontWeight--bold u-textColor--primary u-marginBottom--5">
+              Error description
+            </p>
+            <p className="u-fontSize--small u-textColor--error">
+              {typeof errorMessage === "object"
+                ? "An unknown error orrcured while trying to upload your license. Please try again."
+                : errorMessage}
+            </p>
+            {!size(licenseExistErrData) ? (
+              <div className="flex flex-column">
+                <p className="u-fontSize--small u-fontWeight--bold u-marginTop--15 u-textColor--primary">
+                  Run this command to generate a support bundle
+                </p>
+                <CodeSnippet
+                  language="bash"
+                  canCopy={true}
+                  onCopyText={
+                    <span className="u-textColor--success">
+                      Command has been copied to your clipboard
+                    </span>
+                  }
+                >
+                  kubectl support-bundle https://kots.io
+                </CodeSnippet>
+              </div>
+            ) : (
+              <div className="flex flex-column">
+                <p className="u-fontSize--small u-fontWeight--bold u-marginTop--15 u-textColor--primary">
+                  Run this command to remove the app
+                </p>
+                <CodeSnippet
+                  language="bash"
+                  canCopy={true}
+                  onCopyText={
+                    <span className="u-textColor--success">
+                      Command has been copied to your clipboard
+                    </span>
+                  }
+                >
+                  {typeof licenseExistErrData === "string"
```

Comment:
added a type guard here. we will need to make sure we refactor this kind of thing out. shouldn't be assigning strings and objects to the same variable.